src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -110,12 +110,12 @@ public override Task<bool> HandleRequestAsync()
             var remoteSignOutContext = new RemoteSignOutContext(Context, Scheme, Options, message);
             await Events.RemoteSignOut(remoteSignOutContext);
 
-            if (remoteSignOutContext.HandledResponse)
+            if (remoteSignOutContext.State == EventResultState.HandleResponse)
             {
                 Logger.RemoteSignOutHandledResponse();
                 return true;
             }
-            if (remoteSignOutContext.Skipped)
+            if (remoteSignOutContext.State == EventResultState.SkipToNextMiddleware)
```

Comment:
> Oh I think I see the problem, I missed the fact that we were now pushing these concepts onto all of our middleware, today only OIDC + JwtBearer expose Control contexts... yeah so we don't want to spread this out.

That's absolutely not the objective of this PR and I believe the current state reflects it (I will submit a separate issue/PR so we can discuss making the events consistent across the stack, but that's a totally different story).

This PR has only two goals: adding the options/scheme to the base context class and removing `HandleResponse`/`SkipToNextMiddleware` from every context where it doesn't make any sense to replace it by a `BypassDefaultLogic` concept that does exactly the same thing as in 1.0/1.1 but with a better name and a way better intellisense story.